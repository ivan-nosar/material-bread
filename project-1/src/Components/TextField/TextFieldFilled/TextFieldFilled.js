import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View, TextInput, Platform } from 'react-native';
import withTheme from '../../../Theme/withTheme';
import TextFieldUnderline from '../TextFieldUnderline/TextFieldUnderline';
import TextFieldLabel from '../TextFieldLabel/TextFieldLabel';
import TextFieldHelperText from '../TextFieldHelperText/TextFieldHelperText';
import styles from './TextFieldFilled.styles';

class TextFieldFilled extends Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    style: PropTypes.object,
    containerStyle: PropTypes.object,
    error: PropTypes.bool,
    label: PropTypes.string,
    labelColor: PropTypes.string,
    labelStyle: PropTypes.object,
    handleFocus: PropTypes.func,
    handleBlur: PropTypes.func,
    focused: PropTypes.bool,
    helperText: PropTypes.string,
    helperVisible: PropTypes.bool,
    helperTextStyle: PropTypes.object,
    underlineColor: PropTypes.string,
    underlineActiveColor: PropTypes.string,
    leadingIcon: PropTypes.node,
    trailingIcon: PropTypes.node,
    dense: PropTypes.bool,
    value: PropTypes.bool,
    multiline: PropTypes.bool,
    suffix: PropTypes.node,
    prefix: PropTypes.node,
  };

  static defaultProps = {
    helperVisible: true,
  };

  state = {
    height: 56,
  };

  componentDidUpdate(prevProps) {
    const { value, multiline } = this.props;
    if (value && value.length < 1 && prevProps.value.length > 0 && multiline) {
      this.setState({ height: 56 });
    }
  }

  _renderLeadingIcon() {
    const { leadingIcon } = this.props;

    return (
      <View style={{ position: 'absolute', left: 8, top: 16, zIndex: 200 }}>
        {React.cloneElement(leadingIcon, {
          size: leadingIcon.props.size ? leadingIcon.props.size : 24,
          zIndex: 200,
        })}
      </View>
    );
  }

  _renderTrailingIcon() {
    const { trailingIcon } = this.props;

    return (
      <View style={{ position: 'absolute', right: 12, top: 16 }}>
        {React.cloneElement(trailingIcon, {
          size: trailingIcon.props.size ? trailingIcon.props.size : 24,
        })}
      </View>
    );
  }
  _renderPrefix() {
    const { prefix } = this.props;

    return (
      <View style={{ position: 'absolute', left: 16, top: 26 }}>
        {React.cloneElement(prefix, {
          color: prefix.props.color ? prefix.props.color : 'rgba(0, 0, 0, .57)',
          fontSize: prefix.props.fontSize ? prefix.props.fontSize : 16,
        })}
      </View>
    );
  }

  _renderSuffix() {
    const { suffix } = this.props;

    return (
      <View style={{ position: 'absolute', right: 16, top: 28 }}>
        {React.cloneElement(suffix, {
          color: suffix.props.color ? suffix.props.color : 'rgba(0, 0, 0, .57)',
          fontSize: suffix.props.fontSize ? suffix.props.fontSize : 14,
        })}
      </View>
    );
  }

  _updateTextInputHeight = e => {
    if (!this.props.multiline) return;

    const nativeHeight = e.nativeEvent.contentSize.height;

    this.setState({
      height: nativeHeight < 56 ? 56 : nativeHeight,
    });
  };

  render() {
    const {
      style,
      containerStyle,
      error,
      label,
      labelColor,
      labelStyle,
      handleFocus,
      handleBlur,
      focused,
      helperText,
      helperVisible,
      helperTextStyle,
      underlineColor,
      underlineActiveColor,
      leadingIcon,
      trailingIcon,
      dense,
      suffix,
      prefix,
      ...rest
    } = this.props;

    let height =
      rest.multiline || rest.numberOfLines > 1 ? this.state.height : 56;
    let paddingTop = rest.multiline ? 24 : 12;
    if (dense) {
      height = label ? 52 : 40;
      paddingTop = 6;
    }

    let paddingLeft = leadingIcon ? 44 : 12;
    if (prefix) paddingLeft = 32;

    const platformStyles = Platform.OS == 'web' ? { outlineWidth: 0 } : {};
    return (
      <View
        style={[
          styles.containerStyle,
          { marginBottom: helperText && helperVisible ? 20 : 0 },
          containerStyle,
        ]}>
        <TextFieldLabel
          label={label}
          focused={focused}
          error={error}
          value={rest.value && rest.value.length > 0}
          labelColor={labelColor}
          style={labelStyle}
          leadingIcon={leadingIcon}
          dense={dense}
          prefix={prefix}
          type={'filled'}
        />
        {leadingIcon ? this._renderLeadingIcon() : null}
        {prefix ? this._renderPrefix() : null}
        <TextInput
          style={[
            styles.textField,
            styles.filledInput,
            platformStyles,
            {
              minHeight: dense ? 40 : 56,
              height: height,
              paddingBottom: rest.multiline ? 8 : 0,
              paddingTop: paddingTop,
              paddingLeft: paddingLeft,
              paddingRight: trailingIcon || suffix ? 36 : 0,
            },
            style,
          ]}
          onContentSizeChange={e => this._updateTextInputHeight(e)}
          onFocus={handleFocus}
          onBlur={handleBlur}
          {...rest}
        />
        {trailingIcon ? this._renderTrailingIcon() : null}
        {suffix ? this._renderSuffix() : null}

        <TextFieldUnderline
          focused={focused}
          error={error}
          underlineColor={underlineColor}
          underlineActiveColor={underlineActiveColor}
        />

        <TextFieldHelperText
          error={error}
          visible={helperVisible || error}
          style={helperTextStyle}>
          {helperText}
        </TextFieldHelperText>
      </View>
    );
  }
}

export default withTheme(TextFieldFilled);
