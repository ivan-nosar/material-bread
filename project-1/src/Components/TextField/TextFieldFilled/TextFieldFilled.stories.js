import React from 'react';
import { View } from 'react-native';

import { State, Store } from '@sambego/storybook-state';
import Header from '../../../storybook/components/Header';
import Container from '../../../storybook/components/Container';
import { storiesOf } from '../../../storybook/helpers/storiesOf';
import { TextField, Icon, Caption, BodyText } from '../../../';

const store = new Store({
  filledOne: '',
  filledTwo: '',
  filledThree: '',
});

export default storiesOf('Components|TextField/Filled', module)
  .addParameters({ jest: ['TextFieldFilled'] })

  .add('simple', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Label'}
              value={state.filledOne}
              onChangeText={value => store.set({ filledOne: value })}
            />

            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Label'}
              error={true}
              value={state.filledThree}
              onChangeText={value => store.set({ filledThree: value })}
              helperText={'There is an error'}
            />
          </View>
        )}
      </State>
    </Container>
  ))
  .add('multiline', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Multiline'}
              multiline
              value={state.filledTwo}
              onChangeText={value => store.set({ filledTwo: value })}
            />
          </View>
        )}
      </State>
    </Container>
  ))
  .add('with leadingIcon', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Icon'}
              leadingIcon={
                <Icon name={'date-range'} size={24} color={'#6e6e6e'} />
              }
              value={state.filledTwo}
              onChangeText={value => store.set({ filledTwo: value })}
            />
          </View>
        )}
      </State>
    </Container>
  ))
  .add('with trailingIcon', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Icon'}
              trailingIcon={
                <Icon name={'date-range'} size={24} color={'#6e6e6e'} />
              }
              value={state.filledTwo}
              onChangeText={value => store.set({ filledTwo: value })}
            />
          </View>
        )}
      </State>
    </Container>
  ))
  .add('with prefix', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Pounds'}
              prefix={<BodyText text={'$'} />}
              value={state.filledTwo}
              onChangeText={value => store.set({ filledTwo: value })}
            />
          </View>
        )}
      </State>
    </Container>
  ))
  .add('with suffix', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Pounds'}
              suffix={<Caption text="lbs" />}
              value={state.filledTwo}
              onChangeText={value => store.set({ filledTwo: value })}
            />
          </View>
        )}
      </State>
    </Container>
  ))
  .add('with helperText', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Label'}
              value={state.filledOne}
              onChangeText={value => store.set({ filledOne: value })}
              helperText={'Some helper text'}
            />

            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'Label'}
              error={true}
              value={state.filledThree}
              onChangeText={value => store.set({ filledThree: value })}
              helperText={'This is the error text that will show'}
            />
          </View>
        )}
      </State>
    </Container>
  ))
  .add('dense', () => (
    <Container>
      <Header title={'Filled TextField'} />
      <State store={store} style={{ flex: 1 }}>
        {state => (
          <View>
            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              label={'With Label'}
              value={state.filledOne}
              onChangeText={value => store.set({ filledOne: value })}
              dense
            />

            <TextField
              type={'filled'}
              containerStyle={{ marginTop: 20 }}
              value={state.filledThree}
              onChangeText={value => store.set({ filledThree: value })}
              dense
            />
          </View>
        )}
      </State>
    </Container>
  ));
