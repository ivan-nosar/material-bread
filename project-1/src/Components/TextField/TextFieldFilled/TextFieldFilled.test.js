import React from 'react';
import TextFieldFilled from './TextFieldFilled';

import renderer from 'react-test-renderer';

test('TextFieldFilled Renders', () => {
  const tree = renderer.create(<TextFieldFilled />).toJSON();
  expect(tree).toMatchSnapshot();
});
