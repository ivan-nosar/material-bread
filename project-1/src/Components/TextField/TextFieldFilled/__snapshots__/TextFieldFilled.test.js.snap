// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldFilled Renders 1`] = `
<View
  style={
    Array [
      Object {
        "position": "relative",
      },
      Object {
        "marginBottom": 0,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "rgba(0, 0, 0, 0.54)",
        "marginLeft": 0,
        "paddingHorizontal": 0,
        "position": "absolute",
        "transform": Array [
          Object {
            "scale": 1,
          },
          Object {
            "translateY": 20,
          },
          Object {
            "translateX": 11,
          },
        ],
        "zIndex": 10,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    onContentSizeChange={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "paddingHorizontal": 12,
          "textAlignVertical": "bottom",
        },
        Object {
          "backgroundColor": "#d9d9d9",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {},
        Object {
          "height": 56,
          "minHeight": 56,
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 0,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
    theme={
      Object {
        "action": Object {
          "active": "rgba(0, 0, 0, 0.54)",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "disabledBackground": "rgba(0, 0, 0, 0.12)",
          "hover": "rgba(0, 0, 0, 0.08)",
          "hoverOpacity": 0.08,
          "selected": "rgba(0, 0, 0, 0.14)",
        },
        "background": Object {
          "default": "#fafafa",
        },
        "bodyText": Object {
          "fontSize": 16,
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        "bodyTextTwo": Object {
          "fontSize": 14,
          "letterSpacing": 0.25,
          "lineHeight": 14,
        },
        "button": Object {
          "borderRadius": 4,
          "padding": 8,
        },
        "buttonText": Object {
          "color": "#2196f3",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.5,
        },
        "caption": Object {
          "fontSize": 12,
          "letterSpacing": 0.4,
          "lineHeight": 12,
        },
        "containedButton": Object {
          "backgroundColor": "#2196f3",
          "paddingHorizontal": 16,
        },
        "divider": Object {
          "backgroundColor": "rgba(0,0,0,.12)",
          "width": "100%",
        },
        "error": Object {
          "dark": "#d32f2f",
          "light": "#e57373",
          "main": "#f44336",
        },
        "headingFive": Object {
          "fontSize": 24,
          "lineHeight": 24,
        },
        "headingFour": Object {
          "fontSize": 34,
          "letterSpacing": 0.25,
          "lineHeight": 34,
        },
        "headingOne": Object {
          "fontSize": 96,
          "fontWeight": "300",
          "letterSpacing": -1.5,
          "lineHeight": 96,
        },
        "headingSix": Object {
          "fontSize": 20,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 20,
        },
        "headingThree": Object {
          "fontSize": 48,
          "lineHeight": 48,
        },
        "headingTwo": Object {
          "fontSize": 60,
          "fontWeight": "300",
          "letterSpacing": -0.5,
          "lineHeight": 60,
        },
        "outlinedButton": Object {
          "borderColor": "rgba(0,0,0,.29)",
          "borderWidth": 0.5,
          "paddingHorizontal": 16,
        },
        "overline": Object {
          "fontSize": 10,
          "letterSpacing": 1.5,
          "lineHeight": 10,
          "textTransform": "uppercase",
        },
        "paper": Object {
          "backgroundColor": "white",
          "borderRadius": 2,
        },
        "primary": Object {
          "dark": "#005cb2",
          "light": "#6ab7ff",
          "main": "#1e88e5",
        },
        "secondary": Object {
          "dark": "#c51162",
          "light": "#ff4081",
          "main": "#f50057",
        },
        "subtitleOne": Object {
          "fontSize": 16,
          "letterSpacing": 0.15,
          "lineHeight": 16,
        },
        "subtitleTwo": Object {
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 14,
        },
        "text": Object {
          "disabled": "rgba(0, 0, 0, 0.38)",
          "hint": "rgba(0, 0, 0, 0.38)",
          "primary": "rgba(0, 0, 0, 0.87)",
          "secondary": "rgba(0, 0, 0, 0.54)",
        },
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.42)",
        "bottom": 0,
        "height": 2,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scaleY": 0.5,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(33, 150, 243, 1)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleX": 0,
            },
            Object {
              "scaleY": 0,
            },
          ],
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "bottom": -20,
          "color": "rgba(0,0,0,0.6)",
          "fontSize": 12,
          "height": 16,
          "marginLeft": 12,
          "position": "absolute",
          "width": "100%",
          "zIndex": 10,
        },
        Object {
          "color": "rgba(0,0,0,0.6)",
        },
        undefined,
      ]
    }
  />
</View>
`;
